### PR TITLE
Add Github codeql scanning

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

File: src/libvirtApi/storagePool.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ export function storagePoolCreate({
                 return call(connectionName, storagePoolPath, 'org.freedesktop.DBus.Properties', 'Set', args, { timeout, type: 'ssv' });
             }, exc => {
                 if (storagePoolPath)
-                    storagePoolUndefine(connectionName, storagePoolPath);
+                    storagePoolUndefine({ connectionName, objPath: storagePoolPath });
```
Comment (jelly) [inline src/libvirtApi/storagePool.js:56]: This fixes an issue, but this is probably hard to test.

**Comment by jelly** [comment]
39f6d069343e6dd4907ccf6ff00f82aeb7e11988 broke it

File: src/components/create-vm-dialog/createVmDialog.jsx [hunk 2]
```diff
@@ -1054,7 +1054,7 @@ class CreateVmModal extends React.Component {
             // For different connections the generated VM names might differ
             // try to regenerate it
             if (this.state.os)
-                this.setState({ suggestedVmName: getVmName(value, this.props.vms, this.state.os) });
+                this.setState((_, prevProps) => ({ suggestedVmName: getVmName(value, prevProps.vms, prevProps.os) }));
```
Comment (jelly) [inline src/components/create-vm-dialog/createVmDialog.jsx:1057]: This is wrong..

**Comment by martinpitt** [comment]
The two remaining alerts for 

 > This replaces only the first occurrence of "\n"

are an honest bug. Can you change them to replaceAll() as well, to get this clean?

**Comment by martinpitt** [comment]
Otherwise, good stuff! Let's make the CodeQL scan result green, then land!

**Review by martinpitt — APPROVED** [review]
Perfect, thanks! Nice way to round off this release.

**Pull request merged**